tests/zquantum/qaoa/recursive_qaoa_test.py

```diff
@@ -0,0 +1,125 @@
+from zquantum.core.symbolic_simulator import SymbolicSimulator
+from zquantum.qaoa.ansatzes.farhi_ansatz import QAOAFarhiAnsatz
+from zquantum.qaoa.recursive_qaoa import RecursiveQAOA
+from zquantum.core.estimation import (
+    estimate_expectation_values_by_averaging,
+    allocate_shots_uniformly,
+    calculate_exact_expectation_values,
+)
+import numpy as np
+
+from zquantum.qaoa.problems import (
+    get_maxcut_hamiltonian,
+)
+import networkx as nx
+from zquantum.core.openfermion import change_operator_type
+
+from openfermion import IsingOperator
+from functools import partial
+from zquantum.optimizers import BasinHoppingOptimizer
+import pytest
+
+
+class TestMaxcut:
+    @pytest.fixture()
+    def hamiltonian(self):
+        return (
+            IsingOperator("Z0 Z1", 5)
+            + IsingOperator("Z0 Z3", 5)
+            + IsingOperator("Z1 Z2", 0.5)
+            + IsingOperator("Z2 Z3", 0.5)
+        )
+
+    @pytest.fixture()
+    def backend(self):
+        return SymbolicSimulator()
+
+    def test_rqaoa_returns_correct_answer(self, hamiltonian, backend):
+        # Given
+
+        hmph = RecursiveQAOA(hamiltonian)
+
+        np.random.seed(43)
+        estimation_preprocessors = []
+        estimation_method = calculate_exact_expectation_values
+
+        RQAOA_answers = []
+        optimizer = BasinHoppingOptimizer()
```

Comment:
You don't want to use BasinHoppingOptimizer for the tests, as it introduces dependency on `z-quantum-optimizers`. I don't remember if we have any basic optimizers for tests available in core.

Reply:
Hmm I think there's just a mock optimizer that adds random numbers to parameters each iteration. I'll think of some other tests to test the modularized pieces of logic individually.